### PR TITLE
Improved logging for preference comparison

### PR DESCRIPTION
**Pull request by levmckinney**
This pull request fixes a bug in the logging code for the `EnsembleTrainer` #512. It adds logging of the loss of the ground truth reward on the training data when the ground truth reward is available. This should allow users to more easily check how well the reward model is fitting the training data.

### CODE REVIEW

File: src/imitation/algorithms/preference_comparisons.py [hunk 1]
```diff
@@ -765,10 +780,17 @@ def forward(
...
+        if gt_reward_available:
+            metrics["gt_reward_loss"] = th.nn.functional.binary_cross_entropy(
+                gt_probs,
+                preferences_th,
+            )
+
         return LossAndMetrics(
             loss=th.nn.functional.binary_cross_entropy(probs, preferences_th),
-            metrics={"accuracy": accuracy.detach().cpu()},
+            metrics=metrics,
```
Comment (AdamGleave) [inline src/imitation/algorithms/preference_comparisons.py:793]: Previously we had `.detatch().cpu()`, now we're just returning raw tensors I think. Is this supported?

File: src/imitation/algorithms/preference_comparisons.py [hunk 2]
```diff
@@ -975,17 +997,13 @@ def _training_inner_loop(
         loss = losses.sum()
 
         self.logger.record("loss", loss.item())
-        self.logger.record("loss_std", loss.std().item())
-        # Note here we are returning all the losses not just the mean
-        # This will give us a histogram
-        self.logger.record("dist_loss", losses.detach().cpu().numpy())
```
[the remark this answers is not in the document]
Comment (levmckinney) [reply src/imitation/algorithms/preference_comparisons.py:981]: It doesn't seem to be working, so I decided to remove it.

File: src/imitation/algorithms/preference_comparisons.py (hunk 1 above)
Comment (levmckinney) [reply src/imitation/algorithms/preference_comparisons.py:793]: This could cause crashes if we are training on the GPU or lead to a memory leak. Good catch. I just fixed it.

**Review by AdamGleave — APPROVED** [review]
LGTM
